src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/task/VisitTask.kt

```diff
@@ -36,4 +46,31 @@ class VisitTask(
       visitService.deleteAllExpiredVisitsByApplicationReference(expiredApplicationReferences)
     }
   }
+
+  @Scheduled(cron = "\${task.log-non-associations.cron:0 0 3 * * ?}")
+  fun logNonAssociationVisits() {
+    if (!logNonAssociationsEnabled) {
+      return
+    }
+
+    log.debug("Entered logNonAssociationVisits")
+    prisonConfigService.getSupportedPrisons().forEach { prisonCode ->
+      for (i in 0..numberOfDaysAhead) {
+        val visitFilter = VisitFilter(
+          prisonCode = prisonCode,
+          visitStatusList = listOf(VisitStatus.BOOKED),
+          startDateTime = LocalDateTime.now().plusDays(i).with(LocalTime.MIN),
+          endDateTime = LocalDateTime.now().plusDays(i).with(LocalTime.MAX),
+        )
+        val visits = visitService.findVisitsByFilterPageableDescending(visitFilter)
+
+        visits.forEach {
+          val sessions = sessionService.getVisitSessions(it.prisonCode, it.prisonerId, i, i)
+          if (sessions.isEmpty()) {
+            log.info("Visit with reference - ${it.reference} ,prisoner id - ${it.prisonerId}, prison id - ${it.prisonerId}, start time - ${it.startTimestamp}, end time - ${it.endTimestamp} flagged for check - possible non associations on same day.")
+          }
```

Comment:
May be delay between each iteration,... Thread.sleep()?

Reply:
Added sleep of 1/2 second between visits and updated log message.

Reply:
would also do a try catch 

`    try {
      Thread.sleep(500)
    } catch (e : InterruptedException) {
      log.error("Sleep failed : $e")
    }`